File: baselib/github_base/src/main/java/com/tangpj/github/domain/RepoFlag.java
package com.tangpj.github.domain;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.IntDef;

import static com.tangpj.github.domain.RepoFlag.*;

@IntDef({CREATE,
        STAR,
        FORKED,
        WATCH,
        QUERY})
@Retention(RetentionPolicy.SOURCE)
public @interface RepoFlag{
    int CREATE = 0;
    int STAR = 1;
    int FORKED = 2;
    int WATCH = 3;
    int QUERY = 4;
}

File: baselib/github_base/src/main/java/com/tangpj/github/domain/RepoFlag.kt
package com.tangpj.github.domain

import androidx.annotation.IntDef

@IntDef(value = [RepoFlag.NORMAL, RepoFlag.STAR, RepoFlag.FORKED, RepoFlag.WATCH])
annotation class RepoFlag{
    companion object {
        const val NORMAL = 0
        const val STAR = 1
        const val FORKED = 2
        const val WATCH = 3
    }

}